src/clientiface.cpp

```diff
@@ -277,12 +277,8 @@ void RemoteClient::GetNextBlocks (
 				// Reset usage timer, this block will be of use in the future.
 				block->resetUsageTimer();
 
-				// Block is dummy if data doesn't exist.
-				// It means it has been not found from disk and not generated
-				if(block->isDummy())
-				{
+				if (env->getMap().blockNotInDatabase(p))
```

Comment:
I'm not sure that this is correct.  The block could not be in the database but also not be in the not-exists cache.  Why not just continue using the dummy flag?